expression/builtin_compare.go

```diff
@@ -86,36 +92,181 @@ type coalesceFunctionClass struct {
 	baseFunctionClass
 }
 
-func (c *coalesceFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	if err := c.verifyArgs(args); err != nil {
+func (c *coalesceFunctionClass) getFunction(args []Expression, ctx context.Context) (sig builtinFunc, err error) {
+	// builtinCoalesce returns the first non-NULL value in the list,
+	// or NULL if there are no non-NULL values.
+	// See http://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_coalesce
```

Comment:
move "See ..." this comment to every `builtinCoalesceXXXSig` to be compatible with our existing code.